lib/spack/spack/buildcache_prune.py

```diff
@@ -248,12 +249,108 @@ def _prune_orphans(
     return pruned_object_count
 
 
-def prune(mirror: Mirror, dry_run: bool) -> None:
+def prune_direct(mirror: Mirror, keeplist_file: pathlib.Path, dry_run: bool) -> None:
+    """
+    Execute direct pruning for a given mirror using a keeplist file.
+
+    This function reads a file containing package hashes to keep, then deletes
```

Comment:
you have to be clear what type of hashes you are talking about, package hash means something different in spack.